packages/webgpu/cpp/rnwgpu/SurfaceRegistry.h

```diff
@@ -1,33 +1,159 @@
 #pragma once
 
 #include <memory>
+#include <shared_mutex>
 #include <unordered_map>
 
 #include "webgpu/webgpu_cpp.h"
 
-#include "Canvas.h"
-
 namespace rnwgpu {
 
+struct ISize {
+  int width;
+  int height;
+};
+
+struct SurfaceInfo {
+  void *nativeSurface = nullptr;
+  wgpu::Surface surface;
+  int width;
+  int height;
+  wgpu::Texture texture;
+  wgpu::Instance gpu;
+  wgpu::SurfaceConfiguration config;
+
+  void flush() {
+    // 1.a flush texture to the onscreen surface
```

Comment:
This might be a bit confusing, why does the `SurfaceInfo` struct contain a method `flush` that works only for offscreen texture